include/measurement_kit/dns/query.hpp

```diff
@@ -34,8 +37,8 @@ class Query {
     Query() {}
 
     /// Start an async DNS request.
-    Query(std::string query, std::string address,
-          std::function<void(Response)> func,
+    Query(QueryClass dns_class, QueryType dns_type, std::string name,
```

Comment:
Yes, I think having constructors using string is a good idea. I'll add a diff for that.